lib/galaxy/security/validate_user_input.py

```diff
@@ -65,24 +67,27 @@ def validate_publicname(trans, publicname, user=None):
     return ''
 
 
-def transform_publicname(trans, publicname, user=None):
-    # User names must be at least four characters in length and contain only lower-case
-    # letters, numbers, and the '-' character.
+def transform_publicname(trans, publicname, user=None, check_dup=True):
```

Comment:
`transform_publicname` has two mandatory argument `trans` and `publicname`.  Only the forst one is receive when called from `authenticate`